flink-runtime/src/main/java/org/apache/flink/runtime/shuffle/ShuffleMaster.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.shuffle;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Intermediate result partition registry to use in {@link org.apache.flink.runtime.jobmaster.JobMaster}.
+ */
+public interface ShuffleMaster<T extends ShuffleDescriptor> {
+	CompletableFuture<T> registerPartitionWithProducer(
```

Comment:
missing javadoc, in particular for the return value

Reply:
I will add class parameter description to class level javadoc, not sure method comment will add too much value to it. The method name already says basically what is happening, unless you think that some specific detail should be mentioned.

Reply:
I think interface methods should get JavaDocs especially if they are public API. Since we want to offer external shuffle service implementations this is the case.